app/org/sagebionetworks/bridge/dao/SubpopulationDao.java

```diff
@@ -0,0 +1,76 @@
+package org.sagebionetworks.bridge.dao;
+
+import java.util.List;
+
+import org.sagebionetworks.bridge.models.schedules.ScheduleContext;
+import org.sagebionetworks.bridge.models.studies.Subpopulation;
+import org.sagebionetworks.bridge.models.studies.StudyIdentifier;
+
+public interface SubpopulationDao {
+
+    /**
+     * Create subpopulation. 
+     * @param subpop
+     * @return
+     */
+    public Subpopulation createSubpopulation(Subpopulation subpop);
+    
+    /**
+     * If no subpoulation exists, a default subpopulation can be created. This will be called
+     * as part of creating a study, or when an existing study is found to have no subpopulations.
+     * Subpopulations in turn create a default consent document. 
+     * @param study
+     * @return
+     */
+    public Subpopulation createDefaultSubpopulation(StudyIdentifier study);
+    
+    /**
+     * Get all subpopulations defined for this study. It is possible to create a first default
+     * subpopulation if none exists
+     * @param studyId
+     * @param allExistingRecordsOnly
+     *      if true, this call will not create a default subpopulation if none exist, and it will return all 
+     *      records including deleted subpopulations. If false, the opposite: it will create a default 
+     *      subpopulation and it will hide deleted subpopulations.
+     * @return
+     */
+    public List<Subpopulation> getSubpopulations(StudyIdentifier studyId, boolean allExistingRecordsOnly);
```

Comment:
We need to create a subpopulation as a migration step; in the future when creating new studies, it would be a convenience just as it's a convenience today to create a default consent to start with (in the new design, you have to create a default subpop to create a default consent). Put another way, a study isn't valid without these, all sorts of things can break until all these objects are created. 

I can 1) split the boolean up into two arguments, and 2) definitely can stand to be renamed then for clarity. In practice I think both will be true or both will be false.